lib/internal/process/warning.js

```diff
@@ -41,9 +49,15 @@ function setupProcessWarnings() {
     if (!(warning instanceof Error)) {
       throw new TypeError('\'warning\' must be an Error object or string.');
     }
-    if (throwDeprecation && warning.name === 'DeprecationWarning')
+    if (throwDeprecation && warning.name === 'DeprecationWarning') {
       throw warning;
-    else
-      process.nextTick(() => process.emit('warning', warning));
+    } else {
+      if (!warned) {
+        process.nextTick(() => process.emit('warning', warning));
+        return true;
+      } else {
+        return false;
```

Comment:
Is this correct? If `warned`, then return `false`?

Try this:
```js
let warned = false;
warned = process.emitWarning('A Warning', warned);
warned = process.emitWarning('A Warning', warned);
warned = process.emitWarning('A Warning', warned);
```

What would happen? Btw, it would be nice to have that as a test.